shared/middleware/helpers/storage_id.rb

```diff
@@ -63,7 +63,12 @@ def storage_encrypt_app_id(storage_id, app_id)
 
 def storage_id(endpoint)
   return nil if endpoint == 'shared'
+  return -1 if endpoint == 'readonly'
```

Comment:
I am concerned with using special values for this field. Unless Geoffrey disagrees, I think this should be split out into a separate "access" column, or possibly two columns (indicating r/w/x where x = enumerate for "me" and "all").